src/coreclr/vm/threads.cpp

```diff
@@ -2517,6 +2525,13 @@ Thread::~Thread()
 
     // Wait for another thread to leave its loop in DeadlockAwareLock::TryBeginEnterLock
     CrstHolder lock(&g_DeadlockAwareCrst);
+
+#ifdef FEATURE_INTERPRETER
+    if (m_pInterpThreadContext != nullptr)
```

Comment:
The native `Thread` object lifetime is coupled with the managed `Thread` object lifetime currently. The manager thread dies only after it has been collected by the GC. It can take a while for GC to kick in and collect the object.

`m_pInterpThreadContext` holds onto the interpreter stack that can be a lot of memory. It means that if the app creates a lot of a short-lived threads, we would have a lot of these no longer used interpreter stacks around waiting for GC to kick in.

It would be better to clean this up in `OnThreadTerminate`. `OnThreadTerminate` is executed after the thread stopped doing useful work, without waiting for GC to kick in a collect the managed thread.

Reply:
Thanks, moved.